javaparser-core/src/main/javacc/java.jj

```diff
@@ -708,7 +709,7 @@ TOKEN :
           ( // TODO: Could these escape sequences be extracted out?
               "\\"
               (
-                  ["s","n","t","b","r","f","\\","'","\""]
+                  ["s","n","t","b","r","f","\\","'","\"","{"]
```

Comment:
Why did you add this character here?

Reply:
String Templates. They are escaped like this   -  STR."\{someExpression()}"

Reply:
For the information of others, I think this change was made in the definition of 'Escape Sequences'. I'm not sure that this is the most appropriate right for this change.
https://docs.oracle.com/javase/specs/jls/se21/html/jls-3.html#jls-3.10.7

Reply:
String Templates is still preview feature

Reply:
Pushed a revision with of String Template with it's own matcher